metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -753,23 +926,86 @@ def get_dir_to_process(
             sorted_dirs = sorted(
                 iterator,
                 key=functools.cmp_to_key(partitioned_folder_comparator),
-                reverse=True,
+                reverse=not min,
             )
+            folders = []
             for dir in sorted_dirs:
                 if path_spec.dir_allowed(f"{protocol}{bucket_name}/{dir}/"):
-                    return self.get_dir_to_process(
+                    folders_list = self.get_dir_to_process(
                         bucket_name=bucket_name,
                         folder=dir + "/",
                         path_spec=path_spec,
                         protocol=protocol,
+                        min=min,
                     )
-            return folder
-        else:
-            return folder
+                    folders.extend(folders_list)
+                    if not self.source_config.get_all_partitions:
+                        return folders
+            if folders:
+                return folders
+            else:
+                return [f"{protocol}{bucket_name}/{folder}"]
+        return [f"{protocol}{bucket_name}/{folder}"]
+
+    def get_partitions(
+        self,
+        path_spec: PathSpec,
+        bucket: Any,  # Todo: proper type
+        prefix: str,
+        partition: Optional[str] = None,
```

Comment:
Looks like `partition` argument is not passed in codebase. Please remove if not required, in favour of simplicity.